hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/cache/EntityMapWarmer.java

```diff
@@ -0,0 +1,258 @@
+    private final ThreadPoolExecutor threadpool;
+
+    private EntityMapWarmer(
+            Supplier<AccountStorageAdapter> accountsStorageAdapter,
+            Supplier<UniqueTokenMapAdapter> nftsAdapter,
+            Supplier<TokenRelStorageAdapter> tokenRelsAdapter,
+            GlobalDynamicProperties globalDynamicProperties) {
+        this(
+                accountsStorageAdapter,
+                nftsAdapter,
+                tokenRelsAdapter,
+                new ThreadPoolExecutor(
+                        globalDynamicProperties.cacheCryptoTransferWarmThreads(),
+                        globalDynamicProperties.cacheCryptoTransferWarmThreads(),
+                        0L,
+                        TimeUnit.MILLISECONDS,
+                        new LinkedBlockingQueue<>()));
+    }
+
+    @VisibleForTesting
+    EntityMapWarmer(
+            Supplier<AccountStorageAdapter> accountsStorageAdapter,
+            Supplier<UniqueTokenMapAdapter> nftsAdapter,
+            Supplier<TokenRelStorageAdapter> tokenRelsAdapter,
+            ThreadPoolExecutor threadpool) {
+        this.accountsStorageAdapter = accountsStorageAdapter;
+        this.nftsAdapter = nftsAdapter;
+        this.tokenRelsAdapter = tokenRelsAdapter;
+        this.threadpool = threadpool;
+    }
+
+    public static EntityMapWarmer getInstance(
+            Supplier<AccountStorageAdapter> accountsStorageAdapter,
+            Supplier<UniqueTokenMapAdapter> nftsAdapter,
+            Supplier<TokenRelStorageAdapter> tokenRelsAdapter,
+            GlobalDynamicProperties globalDynamicProperties) {
+        if (instance == null) {
+            instance =
+                    new EntityMapWarmer(accountsStorageAdapter, nftsAdapter, tokenRelsAdapter, globalDynamicProperties);
+        }
+        return instance;
+    }
+
+    /**
+     * Warms the cache for select entities found in the given {@code Round}. Any entities not
+     * found will, of course, not be warmed, but instead will essentially be ignored.
+     *
+     * @param round the round to warm the cache for
+     */
+    public void warmCache(Round round) {
+        removeAllTasks();
+        threadpool.execute(() -> doWarmups(round));
+    }
+
+    // "warmTokenObjs" as in "warm the token and its associated token relation objects"
+    // for both the sending and receiving accounts
+    private void warmTokenObjs(AccountID sender, AccountID receiver, long tokenNum, long senderSerialNum) {
+        if (accountsStorageAdapter.get().areOnDisk()
+                && nftsAdapter.get().isVirtual()
+                && tokenRelsAdapter.get().areOnDisk()) {
+
+            threadpool.execute(() -> {
+                // Sender:
+                final var acctMap = accountsStorageAdapter.get().getOnDiskAccounts();
+                final OnDiskAccount account =
+                        Objects.requireNonNull(acctMap).get(EntityNumVirtualKey.from(EntityNum.fromAccountId(sender)));
```

Comment:
Maybe we can add an `EntityNumVirtualKey.fromAccountId()` factory; this is a bit awkward.